### PR TITLE
MM-15842 making post.update() sync by default

### PR DESCRIPTION
**Pull request by carmo-evan**
#### Summary

Makes the Update method of Post sync by default.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/10986

### CODE REVIEW

File: app/post.go [hunk 1]
```diff
@@ -492,7 +492,7 @@ func (a *App) UpdatePost(post *model.Post, safeUpdate bool) (*model.Post, *model
 	}
 
 	if channel.DeleteAt != 0 {
-		err := model.NewAppError("UpdatePost", "api.post.update_post.can_not_update_post_in_deleted.error", nil, "", http.StatusBadRequest)
+		err = model.NewAppError("UpdatePost", "api.post.update_post.can_not_update_post_in_deleted.error", nil, "", http.StatusBadRequest)
 		return nil, err
```
Comment (hanzei) [inline app/post.go]: Nit: Please inline the error declaration here:
```suggestion
		return nil, model.NewAppError("UpdatePost", "api.post.update_post.can_not_update_post_in_deleted.error", nil, "", http.StatusBadRequest)
```

File: store/sqlstore/post_store.go [hunk 2]
```diff
@@ -129,39 +129,37 @@ func (s *SqlPostStore) Save(post *model.Post) store.StoreChannel {
...
+	}
 
-		if _, err := s.GetMaster().Update(newPost); err != nil {
-			result.Err = model.NewAppError("SqlPostStore.Update", "store.sql_post.update.app_error", nil, "id="+newPost.Id+", "+err.Error(), http.StatusInternalServerError)
-		} else {
-			time := model.GetMillis()
-			s.GetMaster().Exec("UPDATE Channels SET LastPostAt = :LastPostAt  WHERE Id = :ChannelId AND LastPostAt < :LastPostAt", map[string]interface{}{"LastPostAt": time, "ChannelId": newPost.ChannelId})
+	if _, err := s.GetMaster().Update(newPost); err != nil {
+		return nil, model.NewAppError("SqlPostStore.Update", "store.sql_post.update.app_error", nil, "id="+newPost.Id+", "+err.Error(), http.StatusInternalServerError)
+	} else {
```
Comment (hanzei) [inline store/sqlstore/post_store.go]: Nit: You can drop the `else` here.

File: app/file.go [hunk 3]
```diff
@@ -275,9 +275,9 @@ func (a *App) MigrateFilenamesToFileInfos(post *model.Post) []*model.FileInfo {
 
 	if newPost := result.Posts[post.Id]; len(newPost.Filenames) != len(post.Filenames) {
 		// Another thread has already created FileInfos for this post, so just return those
-		fileInfos, err := a.Srv.Store.FileInfo().GetForPost(post.Id, true, false)
-		if err != nil {
-			mlog.Error(fmt.Sprintf("Unable to get FileInfos for migrated post, err=%v", err), mlog.String("post_id", post.Id))
+		fileInfos, errFileInfo := a.Srv.Store.FileInfo().GetForPost(post.Id, true, false)
```
Comment (hanzei) [inline app/file.go]: Could you please elaborate why the different variable name is needed here?

**Comment by jespino** [comment]
Merged! Thanks @carmo-evan! :heart:

**Pull request merged**